src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/UnixPkcs12Reader.cs

```diff
@@ -236,7 +236,10 @@ public void Decrypt(SafePasswordHandle password, bool ephemeralSpecified)
             }
             finally
             {
-                password.DangerousRelease();
+                if (hasRef)
+                {
+                    password.DangerousRelease();
```

Comment:
This one, and the one in export reader, are actually both fine because the AddRef was before the try.  But, the if doesn't really hurt.

Reply:
I will move the DangerousAddRef to the try block then.